prometheus_client/multiprocess.py

```diff
@@ -25,6 +27,47 @@
 _db_pattern = re.compile(r"(\w+)_(\d+)\.db")
 
 
+class MetricsCache(object):
+    def __init__(self):
+        self.metrics = []
+        self.last_archive_duration = 0
+        self.lock = RLock()
+
+    def retrieve_metrics(self):
+        with self.lock:
+            return self.metrics
+
+    def write_metrics(self, metrics, time_elapsed=None):
+        with self.lock:
+            self.last_archive_duration = time_elapsed
+            self.metrics = metrics
+
+    def collect(self):
+        yield GaugeMetricFamily(
+            "archive_duration_seconds",
+            "Time taken to collect the latest batch of metrics",
+            value=self.last_archive_duration)
+
+
+
+_metrics_cache = MetricsCache()
+
+
+class InMemoryCollector(object):
+    """
+    A Collector which simply serves statistics collected by the archiver
+    (cleanup_dead_processes())
+    """
+
+    def __init__(self, registry):
+        if registry:
+            registry.register(self)
+            registry.register(_metrics_cache)
```

Comment:
Can you add a comment about why this is also being added as a `Collector`? It wasn't initially clear